charts/consul/templates/gossip-encryption-autogeneration-podsecuritypolicy.yaml

```diff
@@ -0,0 +1,39 @@
+{{- if .Values.global.gossipEncryption.autoGenerate }}
+---
+apiVersion: policy/v1beta1
+kind: PodSecurityPolicy
+metadata:
+  name: {{ template "consul.fullname" . }}-gossip-encryption-autogenerate
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: {{ template "consul.name" . }}
+    chart: {{ template "consul.chart" . }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+  annotations:
+    "helm.sh/hook": pre-install,pre-upgrade
+    "helm.sh/hook-delete-policy": before-hook-creation
```

Comment:
Does this need a `hook-succeeded` delete policy as well like the job? Is there a reason only the job has it?

Reply:
If you add `hook-succeeded` to these, then helm will delete it immediately after creation, since for a psp to succeed you just need to create it. Basically, helm will delete it before the job even starts. This is because there's no way to group different resources into one logical hook in helm, so it actually treats all these as separate hooks, meaning the job and the psp are separate independent helm hooks.

Hope it makes sense.